src/utils/functions/functions.ts

```diff
@@ -22,7 +22,11 @@ export const getFunctionsDir = (
     options: OptionValues
   },
   defaultValue?: string,
-) => options.functions || config.dev?.functions || config.functionsDirectory || defaultValue
+): string | undefined =>
```

Comment:
https://github.com/netlify/cli/pull/7180

I think you've duplicated the fix from 7130 here; same comments apply.